### PR TITLE
refactor: Update edm4hep IO code to use frame model

### PR DESCRIPTION
**Pull request by paulgessinger**
podio switched to recommending this model, and the store based model we used before is deprecated.

### CODE REVIEW

**Comment by paulgessinger** [comment]
Ok I added CMake checks for the podio and edm4hep version. I need to bump our respective versions in our CI images.

File: Examples/Io/EDM4hep/src/EDM4hepSimHitReader.cpp [hunk 1]
```diff
@@ -23,17 +25,16 @@ EDM4hepSimHitReader::EDM4hepSimHitReader(
...
-  m_store.setReader(&m_reader);
 
   m_outputParticles.maybeInitialize(m_cfg.outputParticles);
   m_outputSimHits.initialize(m_cfg.outputSimHits);
 
-  m_eventsRange = std::make_pair(0, m_reader.getEntries());
+  m_eventsRange = std::make_pair(0, m_reader.getEntries("events"));
 
-  m_collections = m_reader.getCollectionIDTable()->names();
+  // m_collections = m_reader.getCollectionIDTable()->names();
```
Comment (andiwand) [inline Examples/Io/EDM4hep/src/EDM4hepSimHitReader.cpp:34]:
```suggestion
```

File: Examples/Io/EDM4hep/src/EDM4hepSimHitReader.cpp [hunk 2]
```diff
@@ -23,17 +25,16 @@ EDM4hepSimHitReader::EDM4hepSimHitReader(
...
 
-  m_mcParticleCollection =
-      &m_store.get<edm4hep::MCParticleCollection>(m_cfg.inputParticles);
+  // m_mcParticleCollection =
+  // &m_store.get<edm4hep::MCParticleCollection>(m_cfg.inputParticles);
```
Comment (andiwand) [inline Examples/Io/EDM4hep/src/EDM4hepSimHitReader.cpp:37]:
```suggestion
```

File: Examples/Io/EDM4hep/src/EDM4hepSimHitReader.cpp [hunk 3]
```diff
@@ -23,17 +25,16 @@ EDM4hepSimHitReader::EDM4hepSimHitReader(
     : m_cfg(config),
       m_logger(Acts::getDefaultLogger("EDM4hepSimHitReader", level)) {
   m_reader.openFile(m_cfg.inputPath);
-  m_store.setReader(&m_reader);
 
   m_outputParticles.maybeInitialize(m_cfg.outputParticles);
   m_outputSimHits.initialize(m_cfg.outputSimHits);
 
-  m_eventsRange = std::make_pair(0, m_reader.getEntries());
+  m_eventsRange = std::make_pair(0, m_reader.getEntries("events"));
```
Comment (andiwand) [inline Examples/Io/EDM4hep/src/EDM4hepSimHitReader.cpp:33]:
```suggestion
```

**Review by andiwand — APPROVED** [review]
looks good. just found some leftovers

**Review by andiwand — APPROVED** [review]

**Pull request merged**